src/gurobi_optimods/opf/converters.py

```diff
@@ -174,19 +174,38 @@ def convert_case_to_internal_format(case_dict):
     if any(gen["bus"] not in bus_ids for gen in case_dict["gen"]):
         raise ValueError("Unknown bus ID referenced in generator bus")
 
+    # For each field we create a key value for use internally.
+    # Note: index != nodeID (bus_i) for buses.
+    # Note: some parts of the code still rely on these indexes being 1..n and
+    # the keys being in ascending order in dictionary iterators. Be very careful
+    # when trying to change this.
+    # Remove isolated buses and their connected branches
+    remove_buses = {bus["bus_i"] for bus in case_dict["bus"] if bus["type"] == 4}
+    if remove_buses:
+        logger.info(f"Removing buses {remove_buses} (bustype=4)")
```

Comment:
Is the result of this that the isolated buses do not appear in the solution at all? We should have a test to make sure they are put back in.

Reply:
I double-checked with the maintainer of PoweModel.jl (https://lanl-ansi.github.io/PowerModels.jl), those buses are excluded from the solution.